stat/lda.go

```diff
@@ -0,0 +1,208 @@
+// Package stat provides generalized statistical functions
+package stat
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// LD is a type for computing and extracting the linear discriminant analysis of a
+// matrix. The results of the linear discriminant analysis are only valid
+// if the call to LinearDiscriminant was successful.
+type LD struct {
+	n, p  int //n = row, p = col
+	k     int
+	ct    []float64  //Constant term of discriminant function of each class
+	mu    *mat.Dense //Mean vectors of each class
+	svd   *mat.SVD
+	ok    bool
+	eigen mat.EigenSym //Eigen values of common variance matrix
+}
+
+// LinearDiscriminant performs a linear discriminant analysis on the
+// matrix of the input data which is represented as an n×p matrix x where each
+// row is an observation and each column is a variable
+//
+// LinearDiscriminant returns whether the analysis was successful
+//
+// @param x is the training samples
+// @param y is the training labels in [0,k)
+// where k is the number of classes
+// @retun ok returns if whether the analysis was successful
```

Comment:
I don't think `godoc` supports any special tags (ie, `@param`, etc), so I would either remove that or replace with the word "parameter". I would move the sentence "LinearDiscriminant returns..." into the preceding paragraph and change it to "It returns...". You can also remove the line with `@return`.